opm/grid/common/ZoltanPartition.cpp

```diff
@@ -176,16 +137,239 @@ zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
+    std::shared_ptr<CombinedGridWellGraph> gridAndWells;
+    MPI_Barrier(cc);
+    if (cc.rank() == root) {
+        int argc=0;
+        char** argv = 0;
+        float ver = 0;
+
+        rc = Zoltan_Initialize(argc, argv, &ver);
+        zz = Zoltan_Create(MPI_COMM_SELF);
+        if ( rc != ZOLTAN_OK )
+        {
+            OPM_THROW(std::runtime_error, "Could not initialize Zoltan!");
+        }
+
+        setDefaultZoltanParameters(zz);
+        Zoltan_Set_Param(zz, "NUM_GLOBAL_PARTS", std::to_string(cc.size()).c_str());
+
+        // For the load balancer one process has the whole grid and
+        // all others an empty partition before loadbalancing.
+        bool partitionIsEmpty     = cc.rank()!=root;
+
+        if( wells )
+        {
+            Zoltan_Set_Param(zz,"EDGE_WEIGHT_DIM","1");
+            gridAndWells.reset(new CombinedGridWellGraph(cpgrid,
+                                                           wells,
+                                                           transmissibilities,
+                                                           partitionIsEmpty,
+                                                           edgeWeightsMethod));
+            Dune::cpgrid::setCpGridZoltanGraphFunctions(zz, *gridAndWells,
+                                                        partitionIsEmpty);
+        }
+        else
+        {
+            Dune::cpgrid::setCpGridZoltanGraphFunctions(zz, cpgrid, partitionIsEmpty);
+        }
+
+        rc = Zoltan_LB_Partition(zz, /* input (all remaining fields are output) */
+                                 &changes,        /* 1 if partitioning was changed, 0 otherwise */
+                                 &numGidEntries,  /* Number of integers used for a global ID */
+                                 &numLidEntries,  /* Number of integers used for a local ID */
+                                 &numImport,      /* Number of vertices to be sent to me */
+                                 &importGlobalGids,  /* Global IDs of vertices to be sent to me */
+                                 &importLocalGids,   /* Local IDs of vertices to be sent to me */
+                                 &importProcs,    /* Process rank for source of each incoming vertex */
+                                 &importToPart,   /* New partition for each incoming vertex */
+                                 &numExport,      /* Number of vertices I must send to other processes*/
+                                 &exportGlobalGids,  /* Global IDs of the vertices I must send */
+                                 &exportLocalGids,   /* Local IDs of the vertices I must send */
+                                 &exportProcs,    /* Process to which I send each of the vertices */
+                                 &exportToPart);  /* Partition to which each vertex will belong */
+
+        // This is very important: by default, zoltan sets numImport to -1,
+        // as we are only running Zoltan in serial.
```

Comment:
Are you sure that the value is different from the parallel version? For both all vertices are on rank 0 before loadbalancing and zoltan only counts objects that move to other ranks.

I would assume you need to set it because you are using it below to create vectors.

Reply:
I tested this. In the beginning, ```numImport``` is set to ```0```, but then after calling zoltan, it gets set to ```-1``` when run in serial (which makes sense, when running in serial, this value is unused). 

Yes, this is set for consistency with the rest of the code.

Reply:
Thanks for the answer